src/Components/Common/TemperatureFormField.tsx

```diff
@@ -1,77 +1,73 @@
-import { useState } from "react";
 import { FormFieldBaseProps } from "../Form/FormFields/Utils";
-import RangeAutocompleteFormField from "../Form/FormFields/RangeAutocompleteFormField";
+import { fahrenheitToCelsius, celsiusToFahrenheit } from "@/Utils/utils";
 import CareIcon from "../../CAREUI/icons/CareIcon";
 import ButtonV2 from "./components/ButtonV2";
-import { fahrenheitToCelsius } from "../../Utils/utils";
 
-type TemperatureUnit = "celsius" | "fahrenheit";
+import TextFormField from "../Form/FormFields/TextFormField";
 
-type Props = FormFieldBaseProps<number> & {
-  placeholder?: string;
+type TemperatureFormFieldProps = FormFieldBaseProps<string> & {
+  unit: "celsius" | "fahrenheit";
+  setUnit: (unit: "celsius" | "fahrenheit") => void;
 };
 
-export default function TemperatureFormField(props: Props) {
-  const [unit, setUnit] = useState<TemperatureUnit>("fahrenheit");
+export default function TemperatureFormField({
+  onChange,
+  unit,
+  setUnit,
```

Comment:
Let's keep the unit state within this component itself. `value` and `onChange` can accept and update values in Fahrenheit itself but render in celsius based on this state.